### ISSUE TITLE
Is there a factory file?

### ISSUE THREAD

**Issue by wernight** [eclipse-che/che#1590]
I find the way to expose a badge or Che factory to be under-exposed (e.g. Codenvy could show Markdown code to include to put a badge).

My question is, is there a `.eclipse-che.json` or similar file which people could write and include under source control? The current model, that file is on Codenvy or Che hosted. This means it doesn't evolve with the code, and one cannot just copy it and use it on its own running Che.

My suggestion (if not already possible) is to move that file under source control and have the starting from factory be like: `https://my-eclipse-che.com/f?git=<path-to-git-repo>` and when Che downloads that repo (using credentials if necessary), I'd look and create a runtime for the `.eclise-che.json`. Kind of like `.editorconfig` or `pom.yml`.

**Comment by wernight** [comment]
I'm not sure if I understand https://codenvy.readme.io/docs/factories correctly, but it looks like it's already possible by `http://beta.codenvy.com/f?url=https://github.com/<repo-url>` and creating a `factory.json`, just not sure I'm reading this well.

**Comment by wernight** [comment]
Almost correct, it should be named `.codenvy.json`, then it works exactly as described. Perfect!

I'd still suggest a badge in Markdown and HTML format to spread the knownledge. Here is an example if someone is looking for it: https://github.com/wernight/scruffy-server

**Comment by TylerJewell** [comment]
Hi @wernight - when you say that you want a badge or HTML to expose it better, can you elaborate a bit more?  We do provide both mechanisms already.  So wondering if we are just making it hard to understand our docs, or if there is another configuration that is needed.

We document markdown, javascript, and html button configurations:
http://codenvy.readme.io/docs/factories#section-factory-button-object

**Comment by wernight** [comment]
I saw it but not initially.

If I use the Factory tab form the home page I see the copy badge (which requires Flash which is being deprecate so it should also show the code or fallback):
![image](https://cloud.githubusercontent.com/assets/145667/16420667/8590a806-3d52-11e6-8e47-fa9a5a8a1646.png)

Initially saw the menu in the editor "Workspace" > "Create Factory..." which does not allow customization and doesn't offer to get a badge:
![image](https://cloud.githubusercontent.com/assets/145667/16420688/9c063894-3d52-11e6-86b8-760f086b4bc3.png)

Overall I find the interface somewhat confusing. It's not the most important issue but I feel it's a bit confusing initially. Also I'm more a new user building Factories now and not just using prebuilt ones. For example:
- I don't feel that a workspace contains one or more projects. It's true that a project may be used in multiple workspaces but that's also weird. Is it one way or the other, why both ways?
- When creating I feel I need to fill in some form that is asking more than I want and morphing depending on my choices. I'd rather just give where the code is (at which point it might offer to use `.codenvy.json` or `.codenvy.dockerfile`); and only after that it may offer to create a runtime (workspace).

Again it's mostly from my initial usage and now I got used to it (mostly).

**Comment by TylerJewell** [comment]
@wernight - I am going to close this issue, as we have updated the docs and they are accurate.  At some point we wil simplify the process of creating new factories from within the dashboard, but that will require a larger specification that will be handled in another issue.
